### ISSUE TITLE
Vybrat providera pro mapy/geocoding

### ISSUE THREAD

**Issue by mpavlase** [jizdoteka/jizdoteka-web#9]
**Geocoding** je transformace textu na místo (našeptávání názvů měst). Dál se bude hodit i nějaký provider pro zobrazení mapy, routování trasy. Mě se osobně hodně líbí Openstreet podklady HERE (kdysi Nokia). Zdarma tam mají 100 000 req/měsíc (asi nejvíc, co jsem zatím našel). Google nabízí 2500 req/den. Je ale třeba myslet na to, že při psaní názvu města každé písmenko je jedním požadavkem a tak to pak může docela naskákat.

Pro porovnání je super:
http://mc.bbbike.org/mc/?lon=16.969019&lat=49.23954&zoom=12&num=2&mt0=osm-roads&mt1=nokia-map&mt2=hike_bike
Čísla vpravo dole 1 až 8 označují počet mapových podkladů, které cheš vidět najednou.

Spíš tady hodím další odkazy, které jsem v této souvislosti našel:
https://switch2osm.org/providers/
js plugin pro OSM http://leafletjs.com/index.html
geocoding plugin pro Leaflet https://github.com/smeijer/L.GeoSearch
geocoding http://services.gisgraphy.com/static/leaflet/index.html
routování [http://map.project-osrm.org](http://map.project-osrm.org/?z=9&center=49.948569%2C16.556396&loc=49.192244%2C16.611338&loc=50.354223%2C15.618439&hl=en&ly=&alt=&df=&srv=)
geocoding http://photon.komoot.de/
https://pypi.python.org/pypi/osmapi
geocoding poskytovaný [Google Maps](https://developers.google.com/maps/documentation/javascript/examples/geocoding-simple)

**Comment by KosiehBarter** [comment]
Napada me jedna vec:
Kdyz mapu nahrajeme az po vyhledani, da se povazovat jeden cely string jako jeden req?
Paklize ano, tak problem z casti vyresen, ale i tech 2500 je na jeden den kriticky malo

**Comment by mpavlase** [comment]
Pozor, jeden o koze a druhý o voze. Pro zobrazení mapy to je v pohodě, ale myslel jsem především to napovídání názvů měst, které bude rychleji spotřebovávat kvótu.

**Comment by KosiehBarter** [comment]
Napovidani mest - to by mel resit prohlizec, pripadne databaze mest (klidne ji nahardcodujme na 13 kraju + "Prágl") a potom mape muzeme poslat string? Nebo to jsem spatne pochopil?

**Comment by mpavlase** [comment]
Ano, napovídání je záležitost JS na straně prohlížeče, ale ty názvy měst budeme potřebovat od někud tahat.
